cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/store/DefaultStore.java

```diff
@@ -637,6 +639,65 @@ public Void apply(AppMds mds) throws Exception {
   }
 
   @Override
+  public void addSchedule(final Id.Program program, final ScheduleSpecification scheduleSpecification) {
+    txnl.executeUnchecked(new TransactionExecutor.Function<AppMds, Void>() {
+      @Override
+      public Void apply(AppMds mds) throws Exception {
+        ApplicationSpecification appSpec = getAppSpecOrFail(mds, program);
+        Map<String, ScheduleSpecification> schedules = Maps.newHashMap(appSpec.getSchedules());
+        schedules.put(scheduleSpecification.getSchedule().getName(), scheduleSpecification);
+        ApplicationSpecification newAppSpec = new AppSpecificationWithChangedSchedules(appSpec, schedules);
+        // TODO: double check this ProgramType.valueOf()
+        replaceAppSpecInProgramJar(program, newAppSpec,
+                                   ProgramType.valueOf(scheduleSpecification.getProgram().getProgramType().name()));
+        mds.apps.updateAppSpec(program.getNamespaceId(), program.getApplicationId(), newAppSpec);
+        return null;
+      }
+    });
+  }
+
+  @Override
+  public void deleteSchedule(final Id.Program program, final SchedulableProgramType programType,
+                             final String scheduleName) {
+    txnl.executeUnchecked(new TransactionExecutor.Function<AppMds, Void>() {
+      @Override
+      public Void apply(AppMds mds) throws Exception {
+        ApplicationSpecification appSpec = getAppSpecOrFail(mds, program);
+        Map<String, ScheduleSpecification> schedules = Maps.newHashMap(appSpec.getSchedules());
+        ScheduleSpecification removed = schedules.remove(scheduleName);
+        if (removed == null) {
+          throw new NoSuchElementException("no such schedule @ account id: " + program.getNamespaceId() +
```

Comment:
Discussed with Sagar, there is an overall push we need to make to improve Exception Handline in CDAP. We will revisit at that time. It is not necessary to make this change now just for one method in the Store interface.